controlplane/kubeadm/controllers/controller.go

```diff
@@ -580,3 +592,44 @@ func (r *KubeadmControlPlaneReconciler) adoptOwnedSecrets(ctx context.Context, k
 
 	return nil
 }
+
+func (r *KubeadmControlPlaneReconciler) remediateUnhealthy(ctx context.Context, controlPlane *internal.ControlPlane) error {
+	if !controlPlane.RemediationAllowed() || !controlPlane.HasUnhealthyMachine() {
```

Comment:
If I'm not wrong, I was expecting also a test on current replicas >= desired replicas